source/includes/elastic-agents/_031-job-completion.md

```diff
@@ -0,0 +1,70 @@
+## Job Completion
+
+This message is a request to the plugin to terminate the agent of the job that has been completed.
+
+<aside class="notice">
+  <strong>Important:</strong> A plugin may, at its discretion, choose to not actually terminate an agent. This could be because the same agent can be used for another job configuration.
+</aside>
+
+<aside class="warning">
+  <strong>Important:</strong> Since this call may potentially make network requests to terminate the agent, the server will, in order to improve performance, use multiple threads to deliver the this message. It is important that plugin implementors ensure that their code is thread-safe.
```

Comment:
Are we using multithreading it?

Reply:
I am not saying that we should not use it. But looking at current implementation I don't see that this call is made using a new thread.

Reply:
Check `create agent` request docs

Reply:
Create agent is actually uses multiple threads look at the usages of https://github.com/gocd/gocd/blob/ea8bee89641992db159629f71d194a011fc6dcf0/base/src/main/java/com/thoughtworks/go/util/SystemEnvironment.java#L189